src/core/control/jobs/ExportTemplate.cpp

```
@@ -0,0 +1,193 @@
+#include "ExportTemplate.h"
+
+#include <cmath>  // for round
+#include <utility>
+
+#include "control/jobs/ProgressListener.h"  // for ProgressListener
+#include "model/Document.h"                 // for Document
+#include "model/XojPage.h"                  // for XojPage
+#include "util/i18n.h"                      // for _
+#include "view/DocumentView.h"              // for DocumentView
+
+ExportTemplate::ExportTemplate(Document* doc, fs::path filePath): doc{doc}, filePath{std::move(filePath)} {
+    ElementRangeVector fullRange;
+    fullRange.emplace_back(0, doc->getPageCount() - 1);
+    setExportRange(fullRange);
+}
+
+ExportTemplate::~ExportTemplate() {}
+
+void ExportTemplate::setExportBackground(const ExportBackgroundType exportBackground) {
+    this->exportBackground = exportBackground;
+}
+
+void ExportTemplate::setProgressListener(ProgressListener* progressListener) {
+    this->progressListener = progressListener;
+}
+
+void ExportTemplate::setExportRange(const PageRangeVector& exportRange) {
+    if (exportRange.empty()) {
```

This should raise an error rather than export the entire document, I think.